Command/ActivateContentTypeCommand.php

```diff
@@ -0,0 +1,147 @@
+        $this
+            ->addArgument(
+                self::ARGUMENT_CONTENTTYPES,
+                InputArgument::IS_ARRAY,
+                sprintf('Optional array of contenttypes to create. Allowed values: [%s]', $fileNames)
+            )
+            ->addOption(
+                self::OPTION_ALL,
+                null,
+                InputOption::VALUE_NONE,
+                sprintf('Make all contenttypes: [%s]', $fileNames)
+            )
+            ->addOption(
+                self::DEACTIVATE,
+                null,
+                InputOption::VALUE_NONE,
+                sprintf('Deactivate contenttypes')
+            )
+            ->addOption(
+                self::FORCE,
+                null,
+                InputOption::VALUE_NONE,
+                sprintf('Activate the contenttypes even if the mapping is not up to date (flagged as draft)')
+            );
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        /** @var array $types */
+        $types = $input->getArgument(self::ARGUMENT_CONTENTTYPES);
+
+        foreach ($types as $type) {
+            try {
+                $contentType = $this->contentTypeService->getByName($type);
+                if ($contentType->getDirty() && !$this->deactivate && !$this->force) {
+                    $this->io->error(sprintf('Content type %s is dirty please update it\'s mapping or use the force flag', $contentType->getName()));
+                    continue;
+                }
+                $contentType->setActive(!$this->deactivate);
+                $this->contentTypeService->persist($contentType);
+                $this->logger->notice($this->deactivate ? 'command.contenttype.deactivate' : 'command.contenttype.activate', [
+                    EmsFields::LOG_OPERATION_FIELD => EmsFields::LOG_OPERATION_UPDATE,
+                    EmsFields::LOG_CONTENTTYPE_FIELD => $contentType->getName(),
+                ]);
+            } catch (\Exception $e) {
+                $this->io->error($e->getMessage());
+            }
+        }
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->io = new SymfonyStyle($input, $output);
+    }
+
+    protected function interact(InputInterface $input, OutputInterface $output)
+    {
+        $this->io->title($this->deactivate ? 'Deactivate contenttypes' : 'Activate contenttypes');
+        $this->io->section('Checking input');
+
+        /** @var array $types */
+        $types = $input->getArgument(self::ARGUMENT_CONTENTTYPES);
+
+        if (!$input->getOption(self::OPTION_ALL) && count($types) == 0) {
+            $this->chooseTypes($input, $output);
+        }
+
+        if ($input->getOption(self::OPTION_ALL)) {
+            $this->optionAll($input);
+        }
+        $this->deactivate = $input->getOption(self::DEACTIVATE);
+        $this->force = $input->getOption(self::FORCE);
```

Comment:
```suggestion
```
We are not interacting with these options, so you can just all getOption in the execute, also remove the 2 private properties 'deactivate' and 'force' on this command class

Reply:
Agree for the force but the deactivate option : see 30930983116d0cff3ff17bcd77bdedb57115827d

Reply:
Oh okay I see, you need the deactivate in the chooseTypes and interact function. Okay I can live with that :)